juju/model.py

```diff
@@ -2157,32 +2157,30 @@ async def destroy_unit(self, unit_id, destroy_storage=False, dry_run=False, forc
         """Destroy units by name.
 
         """
-        connection = self.connection()
-        app_facade = client.ApplicationFacade.from_connection(connection)
-
-        # Get the corresponding unit tag
-        unit_tag = tag.unit(unit_id)
-        if unit_tag is None:
-            log.error("Error converting %s to a valid unit tag", unit_id)
-            return JujuUnitError("Error converting %s to a valid unit tag", unit_id)
-
-        log.debug(
-            'Destroying unit %s', unit_id)
-
-        return await app_facade.DestroyUnit(units=[{
-            'unit-tag': unit_tag,
-            'destroy-storage': destroy_storage,
-            'force': force,
-            'max-wait': max_wait,
-            'dry-run': dry_run,
-        }])
+        await self.destroy_units(unit_id, destroy_storage, dry_run, force, max_wait)
```

Comment:
I think you're right, this call needs to be fixed, but changing the args on the definition would mean changing the API in a run-by, which a lot of users would be upset about. That needs to be done in a major release.

I also realized we're not `return`ing the result here (mostly because this is called only for its side-effect, but nevertheless). I'll fix it, thanks for the heads up 👍